### PR TITLE
fix(events_list): 13936 server error shown after multiple clicks on event card

### PR DESCRIPTION
**Pull request by Akiyamka**
Additionally I fix case with errors when user switch event very fast

image

### CODE REVIEW

File: src/core/api_client/apiClient.ts [hunk 1]
```diff
@@ -252,11 +251,15 @@ export class ApiClient {
       errorMessage = ApiClient.parseError(problem);
     }
 
-    if (!errorsConfig || !errorsConfig.dontShowErrors) {
-      this.notificationService.error({
-        title: this.translationService.t('error'),
-        description: this.translationService.t(errorMessage),
-      });
+    if (problem.kind !== 'canceled') {
+      if (!errorsConfig || !errorsConfig.dontShowErrors) {
```
Comment (dqunbp) [inline src/core/api_client/apiClient.ts]: My suggestion to make this check more readble
```ts
if (!errorsConfig?.dontShowErrors) {
```

Comment (dqunbp) [reply src/core/api_client/apiClient.ts]: One more suggestion)
```ts
if (problem.kind === 'canceled') {
  console.debug('Request was canceled');
} else if (!errorsConfig?.dontShowErrors) {
  this.notificationService.error({
    title: this.translationService.t('error'),
    description: this.translationService.t(errorMessage),
  });
}
```

File: src/features/events_list/components/FullState/FullState.tsx [hunk 2]
```diff
@@ -53,6 +53,17 @@ export function FullState({
     }
   }, [currentEventId, eventsList, virtuoso, setHasUnlistedEvent]);
 
+  const currentEventRef = useRef(currentEventId);
+  currentEventRef.current = currentEventId;
```
Comment (dqunbp) [inline src/features/events_list/components/FullState/FullState.tsx:57]: Why do we need this assignment?

File: src/features/events_list/components/FullState/FullState.tsx [hunk 3]
```diff
@@ -53,6 +53,17 @@ export function FullState({
...
+  const eventClickHandler = useCallback(
+    (id: string) => {
+      if (id !== currentEventRef.current) {
+        onCurrentChange(id);
+      }
+    },
+    [onCurrentChange, currentEventRef],
```
Comment (dqunbp) [inline src/features/events_list/components/FullState/FullState.tsx:64]: Do we need to include it in deps array? 

`currentEventRef` is immutable

File: src/features/events_list/components/FullState/FullState.tsx (hunk 2 above)
Comment (dqunbp) [reply src/features/events_list/components/FullState/FullState.tsx:57]: For assing on each render, not initial only. Got it

File: src/features/events_list/components/FullState/FullState.tsx (hunk 3 above)
Comment (Akiyamka) [reply src/features/events_list/components/FullState/FullState.tsx:64]: actually is no difference are we include refs in deps array or not, but linter more happy when we add =)

File: src/core/api_client/apiClient.ts (hunk 1 above)
Comment (Akiyamka) [reply src/core/api_client/apiClient.ts]: Love it, i make it event more readable after thinking a while
```ts
    if (problem.kind === 'canceled') {
      console.debug('Request was canceled');
    } else if (errorsConfig === undefined || errorsConfig.hideErrors !== true) {
      this.notificationService.error({
        title: this.translationService.t('error'),
        description: this.translationService.t(errorMessage),
      });
    }
```

**Review by dqunbp — APPROVED** [review]

**Pull request merged**